modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -66,61 +66,67 @@ const getRootName = (editor: Editor): string => editor.inline ? editor.getElemen
 const removeUndefined = <T>(obj: T): T => Obj.filter(obj as Record<string, unknown>, (v) => Type.isUndefined(v) === false) as T;
 
 const mkParserSettings = (editor: Editor): DomParserSettings => {
-  const settings = editor.settings;
+  const getOption = editor.options.get;
   const blobCache = editor.editorUpload.blobCache;
 
   return removeUndefined<DomParserSettings>({
-    allow_conditional_comments: settings.allow_conditional_comments,
-    allow_html_data_urls: settings.allow_html_data_urls,
-    allow_svg_data_urls: settings.allow_svg_data_urls,
-    allow_html_in_named_anchor: settings.allow_html_in_named_anchor,
-    allow_script_urls: settings.allow_script_urls,
-    allow_unsafe_link_target: settings.allow_unsafe_link_target,
-    convert_fonts_to_spans: settings.convert_fonts_to_spans,
-    fix_list_elements: settings.fix_list_elements,
-    font_size_legacy_values: settings.font_size_legacy_values,
-    forced_root_block: settings.forced_root_block,
-    forced_root_block_attrs: settings.forced_root_block_attrs,
-    padd_empty_with_br: settings.padd_empty_with_br,
-    preserve_cdata: settings.preserve_cdata,
-    remove_trailing_brs: settings.remove_trailing_brs,
-    inline_styles: settings.inline_styles,
+    allow_conditional_comments: getOption('allow_conditional_comments'),
```

Comment:
Random question unrelated to this code, how does the switch to strings impact autocomplete of available settings? It just occurred to me this might regress some of the integrator usability improvements that came with publishing types.

Reply:
It didn't impact it at all, as you still get autocomplete suggestions for any settings we've defined in the `OptionTypes` file.